### PR TITLE
ci(release): archive Windows artifacts with zip

### PR DESCRIPTION
**Pull request by yuchanns**
Prerequisites for #483.

Also please release a new version after merging. @yetone

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -61,39 +61,54 @@ jobs:
...
+        run: |
+          New-Item -ItemType Directory -Force -Path results
+
+          Copy-Item -Path "target\release\avante_templates.dll" -Destination "results\avante_templates.dll"
+          Copy-Item -Path "target\release\avante_tokenizers.dll" -Destination "results\avante_tokenizers.dll"
+
+          Set-Location -Path results
+
+          $dllFiles = Get-ChildItem -Filter "*.dll" | Select-Object -ExpandProperty Name
+          Compress-Archive -Path $dllFiles -DestinationPath "avante_lib-${{ matrix.os }}-${{ matrix.feature }}.zip"
```
Comment (aarnphm) [inline .github/workflows/release.yaml:92]: why do need a separate branch for windows? afaik the runner also accept bash right?

**Review by aarnphm — COMMENTED** [review]
qq

File: .github/workflows/release.yaml (hunk 1 above)
Comment (yuchanns) [reply .github/workflows/release.yaml:92]: FYI: https://github.com/yetone/avante.nvim/pull/483#issuecomment-2326707354

Comment (aarnphm) [reply .github/workflows/release.yaml:92]: https://github.com/yetone/avante.nvim/actions/runs/10684672838/job/29615675352

Comment (yuchanns) [reply .github/workflows/release.yaml:92]: Nope. External tools will be required to unarchive `tar.gz` in Windows. So we need to use zip and bash on Windows doesn't support zip in GHA.

Comment (yuchanns) [reply .github/workflows/release.yaml:92]: Powershell doesn't have builtin command to unarchive `tar.gz`.

Comment (aarnphm) [reply .github/workflows/release.yaml:92]: haizzz windows 😮‍💨

Comment (aarnphm) [reply .github/workflows/release.yaml:92]: then it seems to me for L79 we should just zip it with tar right?

Comment (yuchanns) [reply .github/workflows/release.yaml:92]: Neither `tar` nor `tar.gz` is supported. Zip the raw artifacts only.

Comment (aarnphm) [reply .github/workflows/release.yaml:92]: i mean 

```yaml
if [ "${{ matrix.os }}" == "windows-latest" ]; then
  zip ... .zip
else
  tar
```

Comment (yuchanns) [reply .github/workflows/release.yaml:92]: zip is not supported when `shell: bash`

**Review by aarnphm — APPROVED** [review]
LGTM, as I don't have access to Windows, but it seems correct to me 😄

**Pull request merged**